### ISSUE TITLE
Allow Project Find Results to be copied to clipboard

### ISSUE THREAD

**Issue by pascalpp** [atom/find-and-replace#416]
When searching a project, it would be nice to be able to copy the results to the clipboard for sharing with other team members, etc.

**Comment by bronson** [comment]
Yes!!  Also so I could use the results in a text buffer.  Re-order the results, search them, etc.

**Comment by benogle** [comment]
> search them

We really should support using the find in buffer view in a project results pane item.

**Comment by ju-Skinner** [comment]
+1

**Comment by sgb-io** [comment]
+1

**Comment by stevenmills** [comment]
+1

**Comment by JPvRiel** [comment]
+1

**Comment by LePhil** [comment]
+1, and thanks to @sgb-io  for the trick, it's so easy to forget about the dev-tools in Atom :+1:

**Comment by brammittendorff** [comment]
+1

**Comment by dbatwa** [comment]
+1

**Comment by freemanjr** [comment]
+1

**Comment by drosenstark** [comment]
+1

**Comment by dsharpen** [comment]
＋1

**Comment by pj-bea** [comment]
+1

**Comment by bijeebuss** [comment]
+1

**Comment by thomas-lee** [comment]
+1

**Comment by JoshSchofield** [comment]
+1

**Comment by MLubinski** [comment]
+1

**Comment by brunnolou** [comment]
+1

**Comment by Robin-front** [comment]
+1

**Comment by ivanmilosavljevic** [comment]
+1

**Comment by kylefritz** [comment]
+1

**Comment by n0ne** [comment]
+1

**Comment by zohozer** [comment]
+1

**Comment by OmeGak** [comment]
> We really should support using the find in buffer view in a project results pane item.

Really looking forward to that.

**Comment by zohozer** [comment]
Please check how it is handling SublimeText this task. Basically it is creating a new file which contains all the "search in project" results so furthermore you can apply new filters to the search result or just copy/paste elsewhere if you need.

**Comment by OmeGak** [comment]
Yes, but SublimeText doesn't allow goto/collapsing results from the "Search in project" tab. I don't think a simple plain text file is the way to go.

**Comment by gnbl** [comment]
+1 
_Find in Buffer_ currently (1.16.0) does not create a search results list as _Find in Project_ can. Copying the matches is not possible. It is not very good for working with data except for the basic Find-And-Replace. I have to resort to Notepad++ to extract specific lines.

**Comment by Swivelgames** [comment]
@pascalpp I'm having this issue as well. So, I landed on this issue, saw it wasn't resolved, and decided to throw a package together.

I literally just threw this together in a few hours. If it helps someone, awesome!

**Install here**: https://atom.io/packages/copy-find-results

It can be invoked via the **Command Palette** or the **Context Menu** if you're perusing a results buffer.

Additionally, you can setup your keymap to call either `copy-find-results:text` or `copy-find-results:json`.

Please write issue tickets in the repo if you have any. I will be very prompt in resolving them: https://github.com/Swivelgames/copy-find-results/issues

**Comment by psmitt** [comment]
Nice job!
Actually I struggled with extracting scope names from the core module's grammar.
I searched for Regex  "name":"[^"]*"  under Path  */grammars/*.json  and with copy-find-results I managed to copy all lines into a tab. Then a repeated search, just in the buffer, with an Alt+Enter gave me the result list onto my clipboard.
Thanks a lot @pascalpp !

**Comment by pascalpp** [comment]
(Pretty sure you mean @Swivelgames : )

**Comment by Swivelgames** [comment]
@psmitt Glad it worked! 😃

( @pascalpp thanks :P )

**Comment by barbaracabral** [comment]
+1

**Comment by ggderas** [comment]
+1

**Comment by Swivelgames** [comment]
@barbaracabral @ggderas Not trying to plug, but it's helped me, and I'm not sure if it's priority for the contributors to implement this type of functionality. As mentioned before, threw something together that seems to work well.

**Install here**: https://atom.io/packages/copy-find-results
**Issues**: https://github.com/Swivelgames/copy-find-results/issues
([**Original Post**](https://github.com/atom/find-and-replace/issues/416#issuecomment-318177630))

Hope it helps.

**Comment by phil-veloso** [comment]
+1

**Comment by Kidaas** [comment]
+1

**Comment by dgboy2000** [comment]
+1

**Comment by lerrigatto** [comment]
+1

**Comment by andrejtest-675** [comment]
> +1
> 
> For anyone looking to do this quick + dirty, you can hack it by copying the output html (devtools), paste into a browser tab, copy as normal

It gave me only a small piece that is displayed and not the part that is above or below and hidden with scroll.

**Comment by abilogos** [comment]
if anyone had troubles with the plugin, it can be done without atom in unix based systems like this:
```
grep -r  "to search" ./path/to/search/* 
```
